5_assignment/index.js

```diff
@@ -0,0 +1,35 @@
+var fs = require('fs');
+var filesArray = ["1.txt", "2.txt", "3.txt", "4.txt", "5.txt"];
+var fileCount = 0;
+
+function readFilePromise (fileName) {
+	return new Promise(function (resolve, reject) {
+		fs.readFile("./files_to_be_read/" + fileName, function(err, data) {
+			if(err) {
+				reject(err);
+			} else {
+			 	printPattern(fileName);
+				resolve(data);
+			}
+		});
+	});
+}
+
+var printPattern = function (data) {
+  console.log(data.toString());
+  console.log(data.toString());
+  console.log("=======================");
+}
+
+function readFile(fileName) {
+	var toBeReturned =  function(){
+		fileCount++; 
+		if(fileCount <= 5) {
+			readFilePromise(fileName).then(readFile(filesArray[fileCount]));
+		}
+	}
+	return toBeReturned;
+}
+
+console.log("=======================");
+readFile(filesArray[fileCount])();
```

Comment:
It works but I don't really like this. Try to improve the logic so you don't have to hack your way like here, i.e. returning a function from inside another function just to use the wrapper as a callback.